### ISSUE TITLE
Generated different clusters for every run

### ISSUE THREAD

**Issue by sevgiozturk** [csn-le/wave_clus#142]
Hi, I'm using Wave_Clus GUI to sort spikes of one channel data from microwire array recording of rat cortex. It generates first figure here: 
![2_fig2print_Ch16_GUI](https://user-images.githubusercontent.com/17386229/63384618-e6c2ed80-c3a7-11e9-8c50-4c5820e17754.png)

For the same raw data, I run batch codes:
Get_spikes([dataPath '.mat']);
Do_clustering(['Ch16_spikes.mat']);
An it generates this:
![2_fig2print_Ch16](https://user-images.githubusercontent.com/17386229/63384809-51742900-c3a8-11e9-960c-e4a2ac6d329d.png)

After a while, when I run the same batch codes, it keeps generating same one cluster result!
![fig2print_Ch16](https://user-images.githubusercontent.com/17386229/63384870-710b5180-c3a8-11e9-903e-ec767355f4ab.png)

I use same parameters for these three cases.
I increased maxtemp = 0.5 to see if clusters were really unseperatable.
Other changed SPC parameters like this : stdmin = 4; stdmax = 15;
I know SPC is stochastic but at least I need to eliminate some noise ! (Cluster 0)
It seems it stucks within the same close-to-begining-temperature value (0.01)!
Can I at least force it to seperate 2-3 clusters ?
Changing max_clus to 2 or 3 does not work!

**Comment by ferchaure** [comment]
A way to force more clusters is to reduce min_clus, but remember that cluster 0, is not always just noise and artefacts, by definitions are spikes too far away from the other clusters

> Other changed SPC parameters like this : stdmin = 4; stdmax = 15;

Just to be clear, these parameters are used in the detection not for the clustering.

I never try, but with your sampling rate maybe use par.scales = 5 will help.

**Comment by ferchaure** [comment]
any updates?

**Comment by sevgiozturk** [comment]
Sorry for late reply but nope! it did not help! These 2 figures from scales=4 (first figure) and scales=5 (second figure) results.
![scales4](https://user-images.githubusercontent.com/17386229/64251961-5990a480-cf22-11e9-9a87-27b8d474a229.png)
![scales5](https://user-images.githubusercontent.com/17386229/64251968-5d242b80-cf22-11e9-99f0-f945ab5c6a32.png)

**Comment by sevgiozturk** [comment]
Btw I ran it using gui

**Comment by ferchaure** [comment]
Have you tried to reduce min_clus?

**Comment by sevgiozturk** [comment]
Yes it was min_clus=10 for both run.

**Comment by sevgiozturk** [comment]
By observing the differences in spike amplitude in raw data, I manually selected spikes from the Cluster 1 and divided them into 3 clusters via brute force. I plan to do this manual processing for that kind of 1-cluster response of wave_clus to me, at least up to we'll figure out why this happens!
Is that temporary solution make sense to you Fernando?
![fig2print_Ch16](https://user-images.githubusercontent.com/17386229/64332646-94a3de00-cfdd-11e9-9678-7bd3fc36ce17.png)

**Comment by ferchaure** [comment]
Is not the best idea... 
Before the automatic selection of clusters at multiple temperatures, the manual curation of wave_clus results was made by looking the clusters that appear on each of the little peaks in the temperature plot using the _Change temperature_ button and choosing just below the peaks. Remember to select before each change of temperature the clusters you like checking the fix _checkbox_ . It will be better if you do that.

**Comment by sevgiozturk** [comment]
Ok good to know, I did not noticed that I can change temperature manually! thanks :) But I think it already returns the best temperature for that specific run! I think it is the best result if it clusters by itself automatically.
Below is the best result generated using gui for the same raw data after trying many times on different days! It seems it seperated spikes with different amplitudes to different clusters which was the idea I tried to do manually. But I know this is just chance! I may not generate it again!

![fig2print_Ch16](https://user-images.githubusercontent.com/17386229/64522997-f927ab80-d303-11e9-8ce4-68243459a2d0.png)

**Comment by ferchaure** [comment]
> But I think it already returns the best temperature for that specific run

Usually the best solution uses more that one temperature. Did you understand the way to fix clusters  and move to another temperature?

> Below is the best result generated using gui for the same raw data after trying many times on different days! It seems it seperated spikes with different amplitudes to different clusters which was the idea I tried to do manually. 

I think I looks better than before, but remember to press the button force to reduce the spikes in the zero class.

**Comment by sevgiozturk** [comment]
> Usually the best solution uses more that one temperature. Did you understand the way to fix clusters and move to another temperature?
Sure, I already tried 'change temperature' but it did not return clusters a way better than before. I did not need to fix clusters via 'fix' checkbox since it could not generated any new good-looking cluster, only some new ones with ~20-30 spikes in it!
 
> I think I looks better than before, but remember to press the button force to reduce the spikes in the zero class.
Actually I hesitate to use it in case of accidentally including noisy-shaped spikes!, since I'm recording from awake behaving animal that raw data naturally includes motion artifacts! But I think I'll give it a try and maybe 'Manual'ly eliminate some noisy-looking spikes from clusters. Thanks for noticing me about this Fernando :)
